test/test-image.sh

```diff
@@ -2,28 +2,33 @@
 
 set -e
 
+GOARCH=$(go env GOARCH)
+
+: ${CONTAINER_CLI:?Need to provide environment variable CONTAINER_CLI}
 : ${IMAGE:?Need to provide environment variable IMAGE}
 : ${EXPECTED_ARCH:?Need to provide variable EXPECTED_ARCH}
 : ${EXPECTED_VERSION:?Need to provide environment variable EXPECTED_VERSION}
 
-IMAGE_ARCH=$(docker inspect ${IMAGE} | jq -r ".[].Architecture")
+echo "CONTAINER CLI ${CONTAINER_CLI}"
+
+IMAGE_ARCH=$(${CONTAINER_CLI} inspect ${IMAGE} | jq -r ".[].Architecture")
 echo "Image architecture ${IMAGE_ARCH}"
 
 if [[ "${IMAGE_ARCH}" != "${EXPECTED_ARCH}" ]] ; then
   echo "Unexpected architecture, got ${IMAGE_ARCH}, expected ${EXPECTED_ARCH}"
   exit 1
 fi
 
-if [[ "${IMAGE_ARCH}" == $(dpkg --print-architecture) ]] ; then
-  CONTAINER_VERSION=$(docker run --rm ${IMAGE} --version)
+if [[ "${IMAGE_ARCH}" == "${GOARCH}" ]] ; then
```

Comment:
as `dpkg` works on Debian based systems only it was not working to run locally on mac, should be the case for other systems too which doesn't have `dpkg` package